pkg/metrics/handler.go

```diff
@@ -1,52 +1,145 @@
 package metrics
 
 import (
+	"context"
+	"fmt"
 	"net/http"
+	"time"
+
+	"github.com/deislabs/smi-metrics/pkg/mesh"
+
+	log "github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/deislabs/smi-sdk-go/pkg/apis/metrics"
-	"github.com/prometheus/client_golang/api"
-	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/go-chi/chi"
 	"github.com/unrolled/render"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // Handler provides the routes required to serve TrafficMetrics
 type Handler struct {
-	client promv1.API
-
-	queries Queries
+	Mesh mesh.Mesh
 
 	render *render.Render
 }
 
 // NewHandler returns a handler that has been initialized to defaults.
-func NewHandler(url, groupVersion string, queries Queries) (*Handler, error) {
-	promClient, err := api.NewClient(api.Config{Address: url})
-	if err != nil {
-		return nil, err
-	}
-
+func NewHandler(meshInstance mesh.Mesh) (*Handler, error) {
 	return &Handler{
-		client:  promv1.NewAPI(promClient),
-		render:  render.New(),
-		queries: queries,
+		Mesh:   meshInstance,
+		render: render.New(),
 	}, nil
 }
 
+func (h *Handler) jsonResponse(w http.ResponseWriter, status int, payload interface{}) {
+	if err := h.render.JSON(w, status, payload); err != nil {
+		log.Errorf("error rendering response: %s", err)
+		http.Error(w, "error rendering response", 500)
+	}
+}
+
+// AddResourceDefinition adds a api resource definition to context for use in
```

Comment:
`addResourceDefinition` as it isn’t exported anymore (and some other export mismatches).